BUG/Maint: numpy additional pre-failures

we have two new pre-failures, AFAICS after the numpy main version was updated

FAILED statsmodels/base/tests/test_generic_methods.py::TestGenericNegativeBinomial::test_zero_collinear
FAILED statsmodels/discrete/tests/test_discrete.py::test_unchanging_degrees_of_freedom

The second test fails most likely because of (assigning a non-scalar array to a single array element in negbin _hessian_nb2
https://github.com/numpy/numpy/issues/10404

The first test has deprecation warnings in a unit test that no warnings should show up

It's not clear to me why negbin _hessian_nb2 does not cause many more test failures.
(I will try to look into this later today)

--- a/statsmodels/discrete/discrete_model.py
+++ b/statsmodels/discrete/discrete_model.py
@@ -3543,7 +3543,7 @@ class NegativeBinomial(CountModel):
                 if j > i:
                     continue
                 hess_arr[i,j] = np.sum(-exog[:,i,None] * exog[:,j,None] *
-                                       const_arr, axis=0)
+                                       const_arr, axis=0).squeeze()
         tri_idx = np.triu_indices(dim, k=1)
         hess_arr[tri_idx] = hess_arr.T[tri_idx]
 

